UH-StudentServices/student_guide
# HUB-736: Show all degree specific content for Siteimprove

This PR adds:

- A secondary sitemap variation for Siteimprove with links to show all degree programme specific content. This means that for each node there are multiple urls with all the different degree programme term id values provided in a `degree_programme` query parameter (which is used by `Drupal\uhsg_active_degree_programme\ActiveDegreeProgrammeService::getTerm()`). This means for each node we now have `1 + [amount of degree terms]` urls (in effect this results into ~100 urls for each node).
- Language prefix compatibility which allows us to divide sitemaps into multiple pages and an index linking to them. By default simple sitemap index files always linked to non language prefixed urls which we couldn't use in current proxied Student guide. Index files became a necessity with the added url amount as the sitemap spec has a max amount of 50000 urls in one sitemap xml. Also that amount of urls in one file would likely have caused some serious performance issues.
- Added a max url amount of 1000 in one file. This seemed like a relatively performant value when testing on my local.

### Testing

- Start up this branch on your local
- Run `gtask` and go to `/app/build` and run `composer install` to make sure you have the latest version of `simple_sitemap`
- Then go to `/app/build/web` and do the usual `drush updb -y && drush cim -y && drush cr`
- Now you should be able to rebuild the new sitemaps with `drush --uri=http://local.guide.student.helsinki.fi simple-sitemap:rebuild-queue && drush --uri=http://local.guide.student.helsinki.fi simple-sitemap:generate` (that `--uri`-parameter is important to make sure the sitemap urls have a proper domain in them)
- Go get some coffee ☕ 
- When the generation job is done go to http://local.guide.student.helsinki.fi/instructions/admin/config/search/simplesitemap . You should see a full progress bar and two sitemap variations (default and siteimprove)
- Both sitemap links should open up a language prefixed url. For my local db the default sitemap has less than 1000 links so there's no index file and paging, but the siteimprove variation should open an index file with links to pages sitemaps. Check a few links to make sure they work as expected.

### Deployment

- Sitemaps will need to be rebuilt after the deployment. You can use the drush commands above, just remember to change the `--uri`-parameter 😊
